src/WebJobs.Script/OutOfProc/Http/HttpFunctionInvokeDispatcher.cs

```diff
@@ -0,0 +1,200 @@
+
+        internal async void InitializeJobhostLanguageWorkerChannelAsync()
+        {
+            await InitializeJobhostLanguageWorkerChannelAsync(0);
+        }
+
+        internal Task InitializeJobhostLanguageWorkerChannelAsync(int attemptCount)
+        {
+            // TODO: Add process managment for http invoker
+            _httpInvokerChannel = _httpInvokerChannelFactory.CreateHttpInvokerChannel(_scriptOptions.RootScriptPath, _metricsLogger, attemptCount);
+            _httpInvokerChannel.StartWorkerProcessAsync().ContinueWith(workerInitTask =>
+                 {
+                     if (workerInitTask.IsCompleted)
+                     {
+                         _logger.LogDebug("Adding http invoker channel. workerId:{id}", _httpInvokerChannel.Id);
+                         // TODO: wait for status api ok
+                         State = FunctionDispatcherState.Initialized;
+                     }
+                     else
+                     {
+                         _logger.LogWarning("Failed to start http invoker process. workerId:{id}", _httpInvokerChannel.Id);
+                     }
+                 });
+            return Task.CompletedTask;
+        }
+
+        public bool IsSupported(FunctionMetadata functionMetadata, string workerRuntime)
+        {
+            return true;
+        }
+
+        public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions)
+        {
+            _functions = functions;
+
+            if (functions == null || functions.Count() == 0)
+            {
+                // do not initialize function dispachter if there are no functions
+                return;
+            }
+
+            State = FunctionDispatcherState.Initializing;
+            await InitializeJobhostLanguageWorkerChannelAsync(0);
+        }
+
+        public Task InvokeAsync(ScriptInvocationContext invocationContext)
+        {
+            return _httpInvokerService.GetInvocationResponse(invocationContext);
+        }
+
+        public async void WorkerError(HttpWorkerErrorEvent workerError)
+        {
+            if (!_disposing)
+            {
+                _logger.LogDebug("Handling WorkerErrorEvent for workerId:{workerId}", workerError.WorkerId);
+                AddOrUpdateErrorBucket(workerError);
+                await DisposeAndRestartWorkerChannel(workerError.WorkerId);
+            }
+        }
+
+        public async void WorkerRestart(HttpWorkerRestartEvent workerRestart)
+        {
+            if (!_disposing)
+            {
+                _logger.LogDebug("Handling WorkerRestartEvent for workerId:{workerId}", workerRestart.WorkerId);
+                await DisposeAndRestartWorkerChannel(workerRestart.WorkerId);
+            }
+        }
+
+        private async Task DisposeAndRestartWorkerChannel(string workerId)
+        {
+                _logger.LogDebug("Disposing channel for workerId: {channelId}", workerId);
```

Comment:
Will this have an issue with multiple concurrent callers? Do we need to synchronize this logic?

Reply:
this is gets triggered by the process exit handler. And for a give process, it only triggers once per worker process